dashboard/app/controllers/projects_controller.rb

```diff
@@ -650,8 +650,10 @@ def limited_gallery?
 
     project = Projects.new(get_storage_id).get(params[:channel_id])
     is_owner = project[:isOwner]
+    is_frozen = project[:frozen]
 
-    return redirect_to "/projects/#{params[:key]}/#{params[:channel_id]}/edit" if is_owner && request.path.ends_with?('/view')
+    return redirect_to "/projects/#{params[:key]}/#{params[:channel_id]}/edit" if is_owner && request.path.ends_with?('/view') && !is_frozen
```

Comment:
Not sure how much it matters if the request ends in view or edit, but might be simpler if we could just have an initial step that always returns the view for a frozen project:

```
return redirect_to "/projects/#{params[:key]}/#{params[:channel_id]}/view" if is_frozen
```

Reply:
I tried this but page failed to load. Seems like `request.path.ends_with?` is needed.

Reply:
ah ok looks like it starts just redirecting in a loop...I guess we need to confirm that the URL you're redirecting to and from are different, which makes sense. Thanks for checking!